src/cdk/v2/destinations/the_trade_desk/utils.js

```diff
@@ -59,13 +60,18 @@ const processRecordInputs = (inputs, destination) => {
   const successMetadata = [];
   const errorResponseList = [];
 
-  const error = new InstrumentationError('Invalid action type');
+  const invalidActionTypeError = new InstrumentationError('Invalid action type');
+  const emptyFieldsError = new InstrumentationError('Fields cannot be empty');
 
   inputs.forEach((input) => {
     const { fields, action } = input.message;
     const isInsertOrDelete = action === 'insert' || action === 'delete';
 
-    if (isInsertOrDelete) {
+    if (!isInsertOrDelete) {
```

Comment:
we can write this like this:
if (!isInsertOrDelete) {
    errorResponseList.push(handleRtTfSingleEventError(input, invalidActionTypeError, {}));
    return;
}

if (isEmptyObject(fields)) {
      errorResponseList.push(handleRtTfSingleEventError(input, emptyFieldsError, {}));
      return;
    }
    
    this bit more readable

Reply:
Agreed!!
Incorporate the suggestions.